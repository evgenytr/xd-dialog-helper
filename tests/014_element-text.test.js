const text = require('../src/main').types.TEXT;
const actions = {
    change: jest.fn(() => undefined),
    values: jest.fn(() => {
    }),
    close: jest.fn(() => {
    }),
    cancel: jest.fn(() => {
    }),
    registerElement: jest.fn(() => undefined)
};

const rendered = text.render('dialog', {
    id: 'hr',
    label: 'Hello World',
    htmlAttributes: {
        someAttr: 'hello'
    }
}, actions);

describe('Text Element', () => {
    describe('render()', () => {
        it('should render correctly', () => {
            expect(rendered).toMatchSnapshot();

            expect(text.render('dialog', {id: 'noHTML'}, actions)).toMatchSnapshot();
        });
    });
    describe('value()', () => {
        it('should correctly return undefined as value', () => {
            expect(text.value({wrapper: document.createElement('p')})).toBeUndefined();
        });
    });
    describe('validate()', () => {
        it('should always return true', () => {
            expect(text.valid({})).toBeTruthy();
        });
    });
    describe('name', () => {
        expect(text.type).toMatchSnapshot();
    });
});
