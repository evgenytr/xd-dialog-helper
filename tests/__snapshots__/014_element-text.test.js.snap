// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `"Text"`;

exports[`Text Element render() should render correctly 1`] = `
Object {
  "input": undefined,
  "props": Object {
    "htmlAttributes": Object {
      "someAttr": "hello",
    },
    "id": "hr",
    "label": "Hello World",
  },
  "type": Object {
    "render": [Function],
    "type": "Text",
    "valid": [Function],
    "value": [Function],
  },
  "wrapper": <p
    id="dialog-hr"
    someattr="hello"
  >
    Hello World
  </p>,
}
`;

exports[`Text Element render() should render correctly 2`] = `
Object {
  "input": undefined,
  "props": Object {
    "id": "noHTML",
  },
  "type": Object {
    "render": [Function],
    "type": "Text",
    "valid": [Function],
    "value": [Function],
  },
  "wrapper": <p
    id="dialog-noHTML"
  >
    undefined
  </p>,
}
`;
